indexer/ai/src/main/java/com/google/cloudsearch/ai/AISkillSet.java

```diff
@@ -0,0 +1,103 @@
+package com.google.cloudsearch.ai;
+
+import com.google.cloudsearch.exceptions.InvalidConfigException;
+import org.apache.log4j.BasicConfigurator;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.log4j.Logger;
+
+/**
+ * Generates a set of skills based on the configuration.
+ */
+public class AISkillSet {
+
+    private List<AISkill> skillSet;
+    private Logger log = Logger.getLogger(AISkillSet.class);
+
+    /**
+     * Constructor
+     * @param skillSet      The aiSkillSet object from the Configuration.
+     * @param schema        CloudSearch Schema
+     * @throws InvalidConfigException   Throws exception if the config in invalid.
+     */
+    public AISkillSet(JSONObject skillSet, JSONObject schema) throws InvalidConfigException {
+        BasicConfigurator.configure();
+        this.parse(skillSet, schema);
+    }
+
+    /**
+     * Set the skill set.
+     * @param skillSet  List of AI Skills
+     */
+    public void setSkillSet(List<AISkill> skillSet) {
+        this.skillSet = skillSet;
+    }
+
+    /**
+     * Returns the set of skills.
+     * @return
+     */
+    public List<AISkill> getSkillSet() {
+        return this.skillSet;
+    }
+
+    /**
+     * Parse the skill configuration and instantiate the AISkill objects.
+     * @param skillSet  The aiSkillSet object from the Configuration.
+     * @param schema    CloudSearch Schema
+     * @throws InvalidConfigException   Throws exception if the config is invalid.
+     */
+    public void parse(JSONObject skillSet, JSONObject schema) throws InvalidConfigException {
+        List<AISkill> skillList = new ArrayList();
+        JSONArray skills = (JSONArray) skillSet.get(Constants.CONFIG_SKILL_SET);
+        if(skills == null)
+            throw new InvalidConfigException("AI Skill Set not specified in configuration.");
+
+        log.info("Skills specified :");
+        for(Object skillObj : skills) {
+            JSONObject nextSkill = (JSONObject) skillObj;
+            String skillName;
+            String[] nameParts;
+            try {
+                skillName = (String) nextSkill.get(Constants.CONFIG_SKILL_NAME);
+                nameParts = skillName.split("\\.");
+                log.info(nameParts[0] + " " + nameParts[1] + " " + nameParts[2]);
+            }
+            catch(Exception e) {
+                throw new InvalidConfigException("Skill name invalid or not specified.");
+            }
```

Comment:
We should be validating the name parts ourselves instead of putting this in a try catch block.

In general, catching an exception and throwing a different one is avoided.

Reply:
done